### PR TITLE
Balder/do not pregenerate reason

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Review by vekterli — CHANGES_REQUESTED** [review]
Overall looks good, but I would really like to see more message meta data emitted in the message tracker status page before pulling this in. Should not add any  overhead, since all the relevant metadata is present without any heap allocs etc already.

File: messagebus/src/vespa/messagebus/routing/routingtable.cpp [hunk 1]
```diff
@@ -30,17 +30,15 @@ RoutingTable::RoutingTable(const RoutingTableSpec &spec) :
...
     }
     for (uint32_t i = 0; i < spec.getNumRoutes(); ++i) {
         Route route;
         const RouteSpec &routeSpec = spec.getRoute(i);
         for (uint32_t j = 0; j < routeSpec.getNumHops(); ++j) {
             route.addHop(Hop(routeSpec.getHop(j)));
         }
-        _routes.insert(std::map<string, Route>::value_type(routeSpec.getName(),
-                                                                route));
+        _routes.insert(std::map<string, Route>::value_type(routeSpec.getName(), std::move(route)));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/routing/routingtable.cpp]: Fixed

File: messagebus/src/vespa/messagebus/routing/routingtable.cpp [hunk 2]
```diff
@@ -30,17 +30,15 @@ RoutingTable::RoutingTable(const RoutingTableSpec &spec) :
 {
     for (uint32_t i = 0; i < spec.getNumHops(); ++i) {
         const HopSpec& hopSpec = spec.getHop(i);
-        _hops.insert(std::map<string, HopBlueprint>::value_type(hopSpec.getName(),
-                                                                     HopBlueprint(hopSpec)));
+        _hops.insert(std::map<string, HopBlueprint>::value_type(hopSpec.getName(), HopBlueprint(hopSpec)));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/routing/routingtable.cpp]: Fixed

File: storage/src/vespa/storage/distributor/activecopy.cpp [hunk 3]
```diff
@@ -20,80 +20,70 @@ namespace std {
...
+    nodeIndex(node),
+    _ideal(0xffff)
+{
+    const BucketCopy* copy = e->getNode(node);
+    assert(copy != 0);
+    _ready = copy->ready();
+    _trusted = copy->trusted();
+    _active = copy->active();
+    for (uint32_t i=0; i<idealState.size(); ++i) {
+        if (idealState[i] == node) _ideal = i;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/activecopy.cpp]: Done

File: storage/src/vespa/storage/distributor/activecopy.h [hunk 4]
```diff
@@ -4,25 +4,27 @@
...
+    vespalib::string getReason() const;
+    friend std::ostream& operator<<(std::ostream& out, const ActiveCopy& e);
 
-    static ActiveList calculate(
-            const std::vector<uint16_t>& idealState,
-            const lib::Distribution&, BucketDatabase::Entry&);
+    static ActiveList calculate(const std::vector<uint16_t>& idealState,
+                                const lib::Distribution&, BucketDatabase::Entry&);
+
+    uint16_t nodeIndex;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/activecopy.h]: Done

File: storage/src/vespa/storage/distributor/activecopy.h [hunk 5]
```diff
@@ -4,25 +4,27 @@
...
-namespace distributor {
+namespace storage::lib { class Distribution; }
+namespace storage::distributor {
 
 class ActiveList;
 
 struct ActiveCopy {
-    uint16_t nodeIndex;
-    vespalib::string reason;
+public:
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/activecopy.h]: Done

**Comment by baldersheim** [comment]
Closing as it is superceeded by #5451

**Pull request closed without merging**